File: backend/src/RealSync.Core/Entities/ConnectedAccount.cs
using System;
using System.Collections.Generic;
using RealSync.Core.Enums;

namespace RealSync.Core.Entities;

/// <summary>
/// Tài khoản liên kết trên các nền tảng xã hội/portal.
/// </summary>
public class ConnectedAccount : BaseEntity
{
    public Guid WorkspaceId { get; set; }
    public string Provider { get; set; } = string.Empty; // e.g. "Facebook", "TikTok", "Zalo", "YouTube", "Website"
    public PublishingChannelType ChannelType { get; set; }
    public string DisplayName { get; set; } = string.Empty;
    public string ExternalAccountId { get; set; } = string.Empty;
    public string? ExternalParentAccountId { get; set; }
    public string? ProfileUrl { get; set; }
    public string? AvatarUrl { get; set; }
    public CredentialStatus Status { get; set; } = CredentialStatus.PendingSetup;
    public string? GrantedScopesJson { get; set; }
    public DateTime? TokenExpiresAt { get; set; }
    public DateTime? LastValidatedAt { get; set; }
    public string? LastErrorCode { get; set; }
    public string? LastErrorMessage { get; set; }

    // Navigation properties
    public OAuthCredential? OAuthCredential { get; set; }
    public ICollection<PublicationJob> PublicationJobs { get; set; } = new List<PublicationJob>();
}

File: frontend/src/types/connectedAccount.ts
export interface ConnectedAccount {
  id: string;
  provider: string;
  channelType: string;
  displayName: string;
  externalAccountId: string;
  externalParentAccountId: string | null;
  profileUrl: string | null;
  avatarUrl: string | null;
  status: string;
  grantedScopesJson: string | null;
  tokenExpiresAt: string | null;
  lastValidatedAt: string | null;
  lastErrorCode: string | null;
  lastErrorMessage: string | null;
  createdAt: string;
}

export interface ConnectedAccountCreateRequest {
  provider: string;
  channelType: string;
  displayName: string;
  externalAccountId: string;
  externalParentAccountId?: string | null;
  profileUrl?: string | null;
  avatarUrl?: string | null;
  accessToken: string;
  refreshToken?: string | null;
  expiresInSeconds?: number | null;
  grantedScopesJson?: string | null;
}

export interface ConnectedAccountReconnectRequest {
  accessToken: string;
  refreshToken?: string | null;
  expiresInSeconds?: number | null;
}

export interface ConnectedAccountAuditLog {
  id: string;
  userId: string | null;
  userEmail: string | null;
  action: string;
  description: string | null;
  ipAddress: string | null;
  createdAt: string;
}
